src/Form/SettingsForm.php

```diff
@@ -200,14 +205,7 @@ public function buildForm(array $form, FormStateInterface $form_state): array {
       '#tree' => TRUE,
     ];
 
-    $options = [];
-    $providers = $this->userManager->getActiveOIDCProviders();
-
-    foreach ($providers as $provider => $value) {
-      // Drupal config cannot contain ".", why I have a different notation for
-      // key and value.
-      $options[$this->escapeProviderId($provider)] = $value;
-    }
+    $options = $this->configHelper->getAllUserProviders();
```

Comment:
For consistence and symmetry, we should probably use the config helper to save the config in `submitForm`.